src/datumaro/experimental/fields.py

```diff
@@ -246,3 +253,88 @@ def image_path_field(semantic: Semantic = Semantic.Default) -> Any:
         ImagePathField instance configured with the given semantic tags
     """
     return ImagePathField(semantic=semantic)
+
+
+@dataclass(frozen=True)
+class LabelField(Field):
+    """
+    Represents a unified label annotation field that supports both single and multi-label scenarios.
+
+    This field automatically detects whether the input is a single label or multiple labels
+    and handles the conversion accordingly:
+    - Single labels: stored as Int32
+    - Multi-labels: stored as List(Int32)
+    """
+
+    semantic: Semantic
+    dtype: Any
+    multi_label: bool = False  # Flag to indicate if this field should handle multi-labels
+
+    def to_polars_schema(self, name: str) -> dict[str, pl.DataType]:
+        """Generate schema based on whether this is single or multi-label."""
+        if self.multi_label:
+            return {name: pl.List(self.dtype)}
+        else:
+            return {name: self.dtype}
+
+    def to_polars(self, name: str, value: Any) -> dict[str, pl.Series]:
+        """Convert label(s) to Polars format, handling both single and multi-label cases."""
+        if value is None:
+            return {name: pl.Series(name, [None], dtype=self.dtype)}
+
+        # Handle single Label object
+        if hasattr(value, "label"):
+            return {name: pl.Series(name, [value.label], dtype=self.dtype)}
```

Comment:
This case is not needed. Apart from special fields like ImageInfo, we want to avoid structs as much as possible.